src/index/IndexBuilderMain.cpp

```diff
@@ -351,6 +352,13 @@ int main(int argc, char** argv) {
       index.createFromFiles(fileSpecifications);
     }
     bool addFromWordsAndDocsFile = !(wordsfile.empty() || docsfile.empty());
+
+    AD_CONTRACT_CHECK(
```

Comment:
AD_CONTRACT_CHECK is for checking invariants that indicate programming bugs when triggered.
This error can be triggered by the user, so use an ordinary exception (i.e. `if(somethingWentWrong) throw std::runtime_error)`).